pkg/component/networking/nodelocaldns/nodelocaldns.go

```diff
@@ -557,3 +566,29 @@ func (n *nodeLocalDNS) SetClusterDNS(dns []string) {
 func (n *nodeLocalDNS) SetDNSServers(servers []string) {
 	n.values.DNSServers = servers
 }
+
+func (n *nodeLocalDNS) SetIPFamilies(ipfamilies []gardencorev1beta1.IPFamily) {
+	n.values.IPFamilies = ipfamilies
+}
+
+func (n *nodeLocalDNS) getIPVSAddress() (ipvsAddress string) {
+	ipFamiliesSet := sets.New[gardencorev1beta1.IPFamily](n.values.IPFamilies...)
+	if ipFamiliesSet.Has(gardencorev1beta1.IPFamilyIPv4) && !ipFamiliesSet.Has(gardencorev1beta1.IPFamilyIPv6) {
+		ipvsAddress = nodelocaldnsconstants.IPVSAddress
+	}
+	if ipFamiliesSet.Has(gardencorev1beta1.IPFamilyIPv6) && !ipFamiliesSet.Has(gardencorev1beta1.IPFamilyIPv4) {
+		ipvsAddress = nodelocaldnsconstants.IPVSIPv6Address
+	}
+	return ipvsAddress
+}
+
+func (n *nodeLocalDNS) getHealthAddress() (healthAddress string) {
+	ipFamiliesSet := sets.New[gardencorev1beta1.IPFamily](n.values.IPFamilies...)
+	if ipFamiliesSet.Has(gardencorev1beta1.IPFamilyIPv4) && !ipFamiliesSet.Has(gardencorev1beta1.IPFamilyIPv6) {
+		return nodelocaldnsconstants.IPVSAddress
+	}
+	if ipFamiliesSet.Has(gardencorev1beta1.IPFamilyIPv6) && !ipFamiliesSet.Has(gardencorev1beta1.IPFamilyIPv4) {
+		healthAddress = fmt.Sprintf("[%s]", nodelocaldnsconstants.IPVSIPv6Address)
+	}
+	return healthAddress
+}
```

Comment:
How about using the below code to reduce duplication?

```suggestion
func (n *nodeLocalDNS) getIPVSAddress() (ipvsAddress string) {
	return n.getAddress(false)
}

func (n *nodeLocalDNS) getHealthAddress() (healthAddress string) {
	return n.getAddress(true)
}

func (n * nodeLocalDNS) getAddress(useIPv6Brackets bool) string {
	ipFamiliesSet := sets.New[gardencorev1beta1.IPFamily](n.values.IPFamilies...)
	if ipFamiliesSet.Has(gardencorev1beta1.IPFamilyIPv4) && !ipFamiliesSet.Has(gardencorev1beta1.IPFamilyIPv6) {
		return nodelocaldnsconstants.IPVSAddress
	}
	if ipFamiliesSet.Has(gardencorev1beta1.IPFamilyIPv6) && !ipFamiliesSet.Has(gardencorev1beta1.IPFamilyIPv4) {
		if useIPv6Brackets {
			return fmt.Sprintf("[%s]", nodelocaldnsconstants.IPVSIPv6Address)
		}
		return nodelocaldnsconstants.IPVSIPv6Address
		}
	}
	return ""
}
```